spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/convert/MappingJdbcConverter.java

```diff
@@ -152,6 +151,11 @@ public SQLType getTargetSqlType(RelationalPersistentProperty property) {
 		return JdbcUtil.targetSqlTypeFor(getColumnType(property));
 	}
 
+	@Override
+	public SQLType getTargetSqlType(Class<?> property) {
```

Comment:
Let me break my change into several parts:

The reason I introduced this method is to optimize the target `SQLType` search. The public  `getTargetSqlType(RelationalPersistentProperty<?> property)` is first resolving the `Class` of the given `property`, and then it searches the `SQLType` for the given `Class`. This method exists mainly if we already know, what `Class` actually backs the property, so we do not need to do discover it over and over again.

Now, let me explain my thoughts on your suggestions. I generally agree with you, that certainly makes sense, but this PR is solving another problem, and this problem of eliminating the incorrect conversions that are applied at the wrong times. See the bugs linked. I think we should do the following - if we recognize, that there is a need to determine the `SQLType` depending on the dialect (from some issue for instance), we're going to do so, but for now, I suggest leaving it as is. 

What are your thoughts on this, @mp911de?

Reply:
My main concern is that we're continuing to add functionality in terms of putting patches around something that is not designed in the right way instead of fixing the core issue.

I think @schauder should decide how to proceed here.